plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -241,6 +250,40 @@ func (c *nodePlugin) admitPodEviction(nodeName string, a admission.Attributes) e
 	}
 }
 
+func (c *nodePlugin) admitPVCStatus(nodeName string, a admission.Attributes) error {
+	switch a.GetOperation() {
+	case admission.Update:
+		oldPVC, ok := a.GetOldObject().(*api.PersistentVolumeClaim)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetOldObject()))
+		}
+
+		newPVC, ok := a.GetObject().(*api.PersistentVolumeClaim)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+
+		// make copies for comparison
+		oldMetadata := oldPVC.ObjectMeta.DeepCopy()
+		newMetadata := newPVC.ObjectMeta.DeepCopy()
+
+		// zero out resourceVersion to avoid comparing differences,
+		// since the new object could leave it empty to indicate an unconditional update
+		oldMetadata.ResourceVersion = ""
+		newMetadata.ResourceVersion = ""
+
+		// ensure no metadata changed. nodes should not be able to relabel, add finalizers/owners, etc
+		if !apiequality.Semantic.DeepEqual(oldMetadata, newMetadata) {
```

Comment:
I tested this and metadata.selfLink is also somehow different between oldMetadata and newMetadata: on old it's blank and on new it's filled in. Is it a bug?!

Reply:
Expected since self link isn't persisted. I plan to switch the way this compares anyway

Reply:
actually, you need to be sending status patches from the kubelet on the specific fields you need to modify, not a straight update (otherwise, a version-skewed kubelet will erase new fields in PVC status)